### PR TITLE
feat(executor): add TSNE encoder for numeric data

### PR DESCRIPTION
**Pull request by anish2197**
Add TSNE support for numeric data.

Stochastic Neighbor Embedding (SNE) starts by converting the high-dimensional Euclidean distances between data points into conditional probabilities that represent similarities. 

Reference : https://scikit-learn.org/stable/modules/generated/sklearn.manifold.TSNE.html

### CODE REVIEW

File: jina/executors/encoders/numeric/tsne.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        if not self.model:
+            self.model = TSNE(n_components=self.output_dim)
+
+    @batching
+    def encode(self, data: 'np.ndarray', *args, **kwargs) -> 'np.ndarray':
+        """
+        :param data: a `B x T` numpy ``ndarray``, `B` is the size of the batch
+        :return: a `B x D` numpy ``ndarray``
+        """
+        _, num_features = data.shape
```
Comment (hanxiao) [inline jina/executors/encoders/numeric/tsne.py:37]:
```suggestion

```

File: jina/executors/encoders/numeric/tsne.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    def __init__(self,
+                 output_dim: int,
+                 *args,
+                 **kwargs):
+        """
+        :param output_dim: the output size.
+        """
+        super().__init__(*args, **kwargs)
+        self.output_dim = output_dim
+        self.model = None
```
Comment (hanxiao) [inline jina/executors/encoders/numeric/tsne.py:24]:
```suggestion

```

File: jina/executors/encoders/numeric/tsne.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    def post_init(self):
+        from sklearn.manifold import TSNE
+        if not self.model:
+            self.model = TSNE(n_components=self.output_dim)
+
```
Comment (hanxiao) [inline jina/executors/encoders/numeric/tsne.py:30]:
```suggestion
        self.model = TSNE(n_components=self.output_dim)

```

**Comment by hanxiao** [comment]
Bad commit message, please refer to:
- [Commit Message Guideline for the First Time Contributor](https://github.com/jina-ai/jina/issues/553)
- [Contributing Guideline](https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md)

**Comment by anish2197** [comment]
Thanks for the pointers. I've made the changes. Should I do a new pull request to re-run the checks?

**Pull request closed without merging**